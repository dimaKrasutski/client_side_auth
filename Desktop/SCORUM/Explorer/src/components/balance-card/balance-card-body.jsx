import React, { Fragment } from 'react';
import PropTypes from 'prop-types';
import { FormattedMessage } from 'react-intl';
import { formatBalanceWithLocale, getBalanceConverter } from '../../helpers/accounts';
import { currencyPropTypes } from '../../helpers/prop-types/currency';

export const BalanceCardBody = ({
  icon, title, balance, locale, currency,
}) => {
  const rate = 1;
  const formatBalanceLocale = formatBalanceWithLocale(locale);
  const convertToCurrency = getBalanceConverter(rate);
  return (
    <Fragment>
      <span className="d-flex align-items-center pb-1_5">
        {icon}
        <span className="ml-1 font-weight-normal">
          <FormattedMessage id={title} />
        </span>
      </span>
      <div>
        <h2 className="mb-0 h2">
          {balance ?
            formatBalanceLocale(balance) :
            <div className="preloader mt-1_5 mb-2 preloader-lg" />
          }
        </h2>
        <h4 className="text-gray-dark-hint">
          {balance && rate ?
            `${formatBalanceLocale(convertToCurrency(balance))} ${currency}` :
            <div className="preloader mb-1_5" />
          }
        </h4>
      </div>
    </Fragment>
  );
};

BalanceCardBody.propTypes = {
  icon: PropTypes.node.isRequired,
  title: PropTypes.string.isRequired,
  balance: PropTypes.number,
  currency: currencyPropTypes.isRequired,
  locale: PropTypes.string.isRequired,
};

BalanceCardBody.defaultProps = {
  balance: null,
};
