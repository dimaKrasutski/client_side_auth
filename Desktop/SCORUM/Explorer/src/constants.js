import * as THEME_CONSTANTS from 'theme/constants';

export const {
  LOCALE_KEYS,
  LANG_KEYS,
  LOCALE,
  LOCALES,
  DEFAULT_LOCALE,
  DEFAULT_LANG,
} = THEME_CONSTANTS;

export const UNDEFINED_ERROR_ID = 'explorer.error.unknown';

export const MIN_FRACTION_DIGITS = 3;
export const MAX_FRACTION_DIGITS = 9;

export const CURRENCY_USD = 'USD';
export const CURRENCY_EUR = 'EUR';
export const ALL_CURRENCY = [CURRENCY_USD, CURRENCY_EUR];
