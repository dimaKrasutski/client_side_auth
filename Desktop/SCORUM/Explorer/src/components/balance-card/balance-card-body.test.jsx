import React from 'react';
import { BalanceCardBody } from './balance-card-body';
import { DEFAULT_LOCALE, CURRENCY_EUR, CURRENCY_USD } from '../../constants';

const trades = {
  [CURRENCY_EUR]: '1.24',
  [CURRENCY_USD]: '1.02',
};

describe('<BalanceCardBody />', () => {
  test('should match snapshot', () => {
    const wrapper = shallow(<BalanceCardBody balance="1250000.200 SCR" currency={CURRENCY_USD} trades={trades} locale={DEFAULT_LOCALE} title="test" icon={<span>Im icon</span>} />);
    expect(wrapper).toMatchSnapshot();
  });

  describe('when no balance', () => {
    test('should match snapshot', () => {
      const wrapper = shallow(<BalanceCardBody locale={DEFAULT_LOCALE} currency={CURRENCY_USD} trades={trades} title="test" icon={<span>Im icon</span>} />);
      expect(wrapper).toMatchSnapshot();
    });
  });

  describe('when no trades', () => {
    test('should match snapshot', () => {
      const wrapper = shallow(<BalanceCardBody locale={DEFAULT_LOCALE} currency={CURRENCY_USD} title="test" icon={<span>Im icon</span>} />);
      expect(wrapper).toMatchSnapshot();
    });
  });
});
