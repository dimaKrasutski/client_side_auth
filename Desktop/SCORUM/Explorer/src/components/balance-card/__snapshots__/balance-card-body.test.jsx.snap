// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BalanceCardBody /> should match snapshot 1`] = `
<React.Fragment>
  <span
    className="d-flex align-items-center pb-1_5"
  >
    <span>
      Im icon
    </span>
    <span
      className="ml-1 font-weight-normal"
    >
      <FormattedMessage
        id="test"
        values={Object {}}
      />
    </span>
  </span>
  <div>
    <h2
      className="mb-0 h2"
    >
      1,250,000.200
    </h2>
    <h4
      className="text-gray-dark-hint"
    >
      1,250,000.200 USD
    </h4>
  </div>
</React.Fragment>
`;

exports[`<BalanceCardBody /> when no balance should match snapshot 1`] = `
<React.Fragment>
  <span
    className="d-flex align-items-center pb-1_5"
  >
    <span>
      Im icon
    </span>
    <span
      className="ml-1 font-weight-normal"
    >
      <FormattedMessage
        id="test"
        values={Object {}}
      />
    </span>
  </span>
  <div>
    <h2
      className="mb-0 h2"
    >
      <div
        className="preloader mt-1_5 mb-2 preloader-lg"
      />
    </h2>
    <h4
      className="text-gray-dark-hint"
    >
      <div
        className="preloader mb-1_5"
      />
    </h4>
  </div>
</React.Fragment>
`;

exports[`<BalanceCardBody /> when no trades should match snapshot 1`] = `
<React.Fragment>
  <span
    className="d-flex align-items-center pb-1_5"
  >
    <span>
      Im icon
    </span>
    <span
      className="ml-1 font-weight-normal"
    >
      <FormattedMessage
        id="test"
        values={Object {}}
      />
    </span>
  </span>
  <div>
    <h2
      className="mb-0 h2"
    >
      <div
        className="preloader mt-1_5 mb-2 preloader-lg"
      />
    </h2>
    <h4
      className="text-gray-dark-hint"
    >
      <div
        className="preloader mb-1_5"
      />
    </h4>
  </div>
</React.Fragment>
`;
